Controllers/Mobile/v1/ProjectsController.cs

```diff
@@ -331,5 +331,59 @@ private async Task<IActionResult> UnLikeProjectAsync(ProjectLike storedProjectLi
             return Ok("Unliked successfully");
         }
 
+        [Authorize(Roles =$"{Constants.USER_TYPE_CLIENT},{Constants.USER_TYPE_FREELANCER}")]
+        [HttpPost("{projectId}/comments")]
+        public async Task<IActionResult> CreateCommentAsync([FromRoute] long projectId, [FromForm] CommentInputDTO commentInputDTO)
+        {
+            if (!ModelState.IsValid)
+                return CustomBadRequest();
+
+            User? authenticatedUser = await userManager.GetUserAsync(HttpContext.User);
+            if (authenticatedUser == null)
+                return Unauthorized(CreateErrorResponse(StatusCodes.Status401Unauthorized.ToString(),
+                    "Unauthorized user"));
+
+            Project? storedProject = await mainAppContext.Projects.FirstOrDefaultAsync(p => p.Id == projectId);
+            if (storedProject == null)
+                return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "Project not found"));
+
+            Comment? comment = new Comment(commentInputDTO, projectId, authenticatedUser.Id);
+
+            if (commentInputDTO.ImageFile != null)
+                comment.ImageUrl = await fileStorageService.SaveAsync(commentInputDTO.ImageFile);
+
+            await mainAppContext.Comments.AddAsync(comment);
+            await mainAppContext.SaveChangesAsync();
+
+            return Ok(CreateSuccessResponse("Commented"));
+        }
+
+        [HttpGet("{projectId}/comments")]
+        public async Task<IActionResult> GetProjectComments([FromRoute] long projectId, [FromQuery] int pageNumber = 0, [FromQuery]int pageSize = 10)
```

Comment:
- the parameter `pageNumber`  should either be renamed to `page`  or the from query  attribute shoud specify that it is from the query param named  `page`, just to provide consistency for frontend devs.
- pageSize's default value should be stored in Constatns.COMMENTS_DEFAULT_PAGE_SIZE.